pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -466,8 +466,36 @@ func maybeRevertToCutoverTimestamp(
 	}
 
 	updateRunningStatus(ctx, j, fmt.Sprintf("starting to cut over to the given timestamp %s", cutoverTime))
+
+	origNRanges := -1
 	spans := []roachpb.Span{sd.Span}
+	updateJobProgress := func() error {
+		if spans == nil {
+			return nil
+		}
```

Comment:
I'm fine with this defensive check, but at the moment we should never have nil here right?

Reply:
Yup it should never be nil.